### PR TITLE
Set CPU automatically when running OpenVINO IR model

### PR DESCRIPTION
**Pull request by kprokofi**
### Summary

Fix
https://github.com/openvinotoolkit/training_extensions/issues/3168
and
cache overriding issue

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Comment by kprokofi** [comment]
https://github.com/openvinotoolkit/training_extensions/issues/3168

File: src/otx/engine/engine.py [hunk 1]
```diff
@@ -311,6 +311,8 @@ def test(
         if is_ir_ckpt and not isinstance(model, OVModel):
             datamodule = self._auto_configurator.update_ov_subset_pipeline(datamodule=datamodule, subset="test")
             model = self._auto_configurator.get_ov_model(model_name=str(checkpoint), label_info=datamodule.label_info)
+            if self.device.accelerator == "auto":
+                self.device = DeviceType.cpu  # type: ignore[assignment]
```
Comment (harimkang) [inline src/otx/engine/engine.py]: If we always need to use the CPU, do we need an if statement?

Comment (jaegukhyun) [reply src/otx/engine/engine.py]: I'm also curious, when user give device gpu by mistake

**Review by jaegukhyun — DISMISSED** [review]

**Comment by harimkang** [comment]
We found cache overriding issue. so fixed - https://github.com/openvinotoolkit/training_extensions/pull/3220/commits/0b9087e9d155d326dac63d5f38fe4e468e23d77b

**Review by harimkang — APPROVED** [review]

**Review by jaegukhyun — APPROVED** [review]

**Comment by harimkang** [comment]
@kprokofi Can you make this change to develop as well?

**Pull request merged**